qiskit_nature/problems/sampling/protein_folding/peptide/beads/base_bead.py

```diff
@@ -0,0 +1,103 @@
+            residue_type: A character representing the type of a residue for the bead.
+            turn_qubits: A list of Pauli operators that encodes the turn following from a
+                            given bead index.
+            build_turn_indicator_fun_0: method that build turn indicator functions for the bead.
+                                        It is passed by a child class (SideBead or MainBead) and
+                                        uses turn qubits to construct a corresponding turn
+                                        indicator function (for details, see the paper: paper
+                                        Robert et al., npj quantum information 7, 38, 2021).
+            build_turn_indicator_fun_1: method that build turn indicator functions for the bead.
+                                        It is passed by a child class (SideBead or MainBead) and
+                                        uses turn qubits to construct a corresponding turn
+                                        indicator function (for details, see the paper: paper
+                                        Robert et al., npj quantum information 7, 38, 2021).
+            build_turn_indicator_fun_2: method that build turn indicator functions for the bead.
+                                        It is passed by a child class (SideBead or MainBead) and
+                                        uses turn qubits to construct a corresponding turn
+                                        indicator function (for details, see the paper: paper
+                                        Robert et al., npj quantum information 7, 38, 2021).
+            build_turn_indicator_fun_3: method that build turn indicator functions for the bead.
+                                        It is passed by a child class (SideBead or MainBead) and
+                                        uses turn qubits to construct a corresponding turn
+                                        indicator function (for details, see the paper: paper
+                                        Robert et al., npj quantum information 7, 38, 2021).
+        """
+        self.chain_type = chain_type
+        self.main_index = main_index
+        self._residue_type = residue_type
+        _validate_residue_symbol(residue_type)
+        self._turn_qubits = turn_qubits
+        if self._residue_type is not None and self.turn_qubits is not None:
+            self._full_id = _build_full_identity(turn_qubits[0].num_qubits)
+            self._turn_indicator_fun_0 = build_turn_indicator_fun_0()
+            self._turn_indicator_fun_1 = build_turn_indicator_fun_1()
+            self._turn_indicator_fun_2 = build_turn_indicator_fun_2()
+            self._turn_indicator_fun_3 = build_turn_indicator_fun_3()
+
+    @property
+    def turn_qubits(self) -> List[PauliOp]:
+        """Returns the list of two qubits that encode the turn following from the bead."""
```

Comment:
I don't fully understand this naming + type + docstring..

* `turn_qubits` suggests we are dealing with qubits
* the docstring supports this but specifically says its 2 qubits, is that right?
* if so, how is the type a `List[PauliOp]`?

Reply:
The are 2 qubits that represent quantum decision variables encoding a type of each turn. But they are each defined as a `PauliOp` which has a size that allows for encoding all of the turns. For example, if there are 3 turns in a peptide, we need 6 qubits (we do some more optimizations but this is irrelevant for my point) and the second turn would be characterized by the following 2 PauliOps: IIZIII, IIIZII, which I store as a list above [IIZIII, IIIZII]. It should be a tuple since we don't modify them later.
Let me know what you think about it.

Reply:
Aha so per turn you have two qubits. A `PauliOp` for a given turn will then have some operations on the qubits associated with this turn and identities on the rest.

So `turn_qubits` is to be understood as "turning the qubits" (i.e. "turn" as a verb) rather than "qubits at which we turn" ("turn" as a description). That makes more sense then :+1: 

I still think that this docstring could be improved though

Reply:
I think they should be understood as an encoding of the direction at which a link between 2 beads points in the lattice. There are 4 possible directions in the model and hence 2 qubits needed to encode them.